specs/eip4844/p2p-interface.md

```diff
@@ -139,45 +155,56 @@ Per `context = compute_fork_digest(fork_version, genesis_validators_root)`:
 | `BELLATRIX_FORK_VERSION` | `bellatrix.SignedBeaconBlock` |
 | `CAPELLA_FORK_VERSION`   | `capella.SignedBeaconBlock`   |
 
-#### BeaconBlockAndBlobsSidecarByRoot v1
+No more than `MAX_REQUEST_BLOCKS_EIP4844` may be requested at a time.
 
-**Protocol ID:** `/eth2/beacon_chain/req/beacon_block_and_blobs_sidecar_by_root/1/`
+#### BlobSidecarsByRoot v1
+
+**Protocol ID:** `/eth2/beacon_chain/req/blob_sidecars_by_root/1/`
 
 Request Content:
 
+```python
+class BlobIdentifier(Container):
+    block_root: Root
+    index: uint64
+```
+
 ```
 (
-  List[Root, MAX_REQUEST_BLOCKS]
+  List[BlobIdentifier, MAX_REQUEST_BLOCKS_EIP4844]
 )
 ```
 
 Response Content:
 
 ```
 (
-  List[SignedBeaconBlockAndBlobsSidecar, MAX_REQUEST_BLOCKS]
+  List[BlobSidecar, MAX_REQUEST_BLOCKS_EIP4844]
 )
 ```
 
-Requests blocks by block root (= `hash_tree_root(SignedBeaconBlockAndBlobsSidecar.beacon_block.message)`).
-The response is a list of `SignedBeaconBlockAndBlobsSidecar` whose length is less than or equal to the number of requests.
+Requests sidecars by block root and index.
+The response is a list of `BlobSidecar` whose length is less than or equal to the number of requests.
 It may be less in the case that the responding peer is missing blocks and sidecars.
 
-No more than `MAX_REQUEST_BLOCKS` may be requested at a time.
+The response is unsigned, i.e. `BlobSidecar`, as the signature of the beacon block proposer
```

Comment:
I agree that these can/should be dropped eventually but you at least need them within the IWANT/IHAVE window, right?

Reply:
IHAVE/IWANT are somewhat unrelated since they sit in the libp2p / gossipsub layer - the gossipsub layer doesn't care about types that way instead dealing with `message id -> message` where message has been type-punned.